Added folder and datacenter to the examples

Tronde opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
From #22644 it seems that the parameter 'folder' is mandatory. While both are required parameters they should be used in the examples.
<!---
If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Docs Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
vmware_guest_snapshot

##### ANSIBLE VERSION
<!--- Paste verbatim output from “ansible --version” between quotes below -->
```
ansible 2.3.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.5 (default, Aug  2 2016, 04:20:16) [GCC 4.8.5 20150623 (Red Hat 4.8.5-4)]
```

Tronde committed:
Added folder and datacenter to the examples

From #22644 it seems that the parameter 'folder' is mandatory. While both are required parameters they should be used in the examples.

gundalow commented on lib/ansible/modules/cloud/vmware/vmware_guest_snapshot.py:
```diff
@@ -60,6 +60,7 @@
    folder:
         description:
             - Define instance folder location.
+        required: True
```
this doesn't look correct
`folder=dict(required=False, type='str', default='/vm'),`

I think it should actually be
`default: /vm` on this line

bcoca replied:
this is not a documentation issue, module should be changed to either require folder or implement an actual search.

Tronde replied:
I agree that the issue with the folder parameter may not be a documentation issue in the first place. I'm also not sure if it makes much sense to define a default value for an optional parameter. Maybe that discussion should stay in #22644.

I suggest to remove line 63 again, but leave the examples as they are. Showing the option in the examples don't hurt. Would you agree on that?

bcoca replied:
optional parameters should always have a default, the module does not need to use it though.

A saner default would have been 'null/None' in which case the module should ignore it.

Tronde committed:
Removed 'required: True' from folder documentation

I reverted the documentation of the folder option to be a not required one. This way it stays consistent with the definition of the function.

Tronde commented:
So I reverted the change to the folder documentation to be a non required option again. This way the documentation is consistent to the code, where it is defined as not required as well:
`folder=dict(required=False, type='str', default='/vm'),`

This way the discussion about how to change this option could take place outside of this PR.

The parameter 'datacenter' should be used in the examples, because it is a mandatory one. This way the examples are consistent with the parameter definition at the beginning of the module documentation. The 'folder' parameter should be used in the examples as well. It doesn't hurt, whether it is required or not. And in the current version you have to use it or your vm won't be found.

Tronde commented:
This PR would fix #24203.

Akasurde reviewed (approved):
shipit

dav1x commented:
shipit
